File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int half = nums.size() / 2;
        unordered_map<int, int> map;
        for (auto n : nums)
            ++map[n];
        for (auto [n, cnt] : map)
        {
            if (cnt > half) return n;
        }
        return -1;
    }
};

File: 0169-majority-element/0169-majority-element.cs
public class Solution {
    public int MajorityElement(int[] nums)
    {
        int major = nums[0];
        int count = 1;

        for (int i = 1; i < nums.Length; i++)
        {
            if (count == 0)
            {
                major = nums[i];
                ++count;
            }
            else if (major == nums[i]) ++count;
            else --count;
        }

        return major;
    }
}